### PR TITLE
Wrap up Brennan's concatenation work.

### PR DESCRIPTION
**Pull request by dabrahams**
Depends on #106, based on #84

### CODE REVIEW

File: Sources/PenguinStructures/Concatenation.swift [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    set {
+      switch i.position {
+      case .a(let index): first[index] = newValue
+      case .b(let index): second[index] = newValue
+      }
+    }
+  }
+}
+
+// TODO: Add MutableCollection conformance.
```
Comment (saeta) [inline Sources/PenguinStructures/Concatenation.swift:192]: I think you can remove this Todo.

File: Sources/PenguinStructures/Concatenation.swift [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+extension Collection {
+  /// Returns a new collection where all the elements of `self` appear before all the elements of
+  /// `other`.
+  @inlinable
+  public func joined<Other: Collection>(to other: Other) -> Concatenation<Self, Other>
```
Comment (saeta) [inline Sources/PenguinStructures/Concatenation.swift:198]: Is joined the right name here? Or should we rename the type?

File: Sources/PenguinStructures/Concatenation.swift (hunk 1 above)
Comment (dabrahams) [inline Sources/PenguinStructures/Concatenation.swift:192]:
```suggestion
```

File: Sources/PenguinStructures/Concatenation.swift (hunk 2 above)
Comment (dabrahams) [reply Sources/PenguinStructures/Concatenation.swift:197]: It's a tough call.  Maybe `concatenated(to:)`.  It was the `with:` that bothered me most as it appeared to disclaim any description of the ordering, and perhaps I took it too far.

https://developer.apple.com/documentation/swift/array/1690077-joined

is significantly different in that it includes a separator and also it produces a `JoinedCollection`, so there's precedent for naming consistency here.

Comment (dabrahams) [reply Sources/PenguinStructures/Concatenation.swift]: Changed.

**Pull request merged**